File: elixir/darts/lib/darts.ex
defmodule Darts do
  @type position :: {number, number}

  @doc """
  Calculate the score of a single dart hitting a target
  """
  @spec score(position) :: integer
  def score({x, y}) do
    d = :math.sqrt(x ** 2 + y ** 2)

    cond do
      d <= 1 -> 10
      d <= 5 -> 5
      d <= 10 -> 1
      true -> 0
    end
  end
end

File: go/darts/darts.go
package darts

import "math"

func Score(x, y float64) int {
	d := math.Sqrt(x*x + y*y)
	switch {
	case d > 10:
		return 0
	case d <= 10 && d > 5:
		return 1
	case d <= 5 && d > 1:
		return 5
	default:
		return 10
	}
}
